orca/3.7-predicting-house-prices.ipynb

```diff
@@ -0,0 +1,1239 @@
+      "\u001b[2m\u001b[36m(pid=30603)\u001b[0m Cause: Unable to locate the source code of <function preprocess at 0x7f602812d7a0>. Note that functions defined in certain environments, like the interactive Python shell do not expose their source code. If that is the case, you should to define them in a .py source file. If you are certain the code is graph-compatible, wrap the call using @tf.autograph.do_not_convert. Original error: could not get source code\n",
+      "\u001b[2m\u001b[36m(pid=30603)\u001b[0m To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert\n",
+      "\u001b[2m\u001b[36m(pid=30603)\u001b[0m WARNING:tensorflow:AutoGraph could not transform <function preprocess at 0x7f5fe867fb90> and will run it as-is.\n",
+      "\u001b[2m\u001b[36m(pid=30603)\u001b[0m Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.\n",
+      "\u001b[2m\u001b[36m(pid=30603)\u001b[0m Cause: Unable to locate the source code of <function preprocess at 0x7f5fe867fb90>. Note that functions defined in certain environments, like the interactive Python shell do not expose their source code. If that is the case, you should to define them in a .py source file. If you are certain the code is graph-compatible, wrap the call using @tf.autograph.do_not_convert. Original error: could not get source code\n",
+      "\u001b[2m\u001b[36m(pid=30603)\u001b[0m To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert\n",
+      "\u001b[2m\u001b[36m(pid=30603)\u001b[0m 2021-01-27 17:53:29.087254: W tensorflow/core/common_runtime/eager/context.cc:566] Unable to destroy server_ object, so releasing instead. Servers don't support clean shutdown.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from zoo.orca.learn.tf2 import Estimator\n",
+    "\n",
+    "k = 4\n",
+    "batch_size = 1\n",
+    "num_epochs = 10\n",
+    "all_mae_histories = []\n",
+    "for i in range(k):\n",
+    "    print('processing fold #', i)\n",
+    "    mae_history = [] # storing all validation MAE\n",
+    "    est = Estimator.from_keras(model_creator=build_model,\n",
+    "                               config={},\n",
+    "                               workers_per_node=1,\n",
+    "                               verbose=0)\n",
+    "    for _ in range(num_epochs):\n",
+    "        stats = est.fit(train_data_creator, \n",
+    "                        epochs=num_epochs,\n",
+    "                        batch_size=batch_size,\n",
+    "                        steps_per_epoch=450 // batch_size,\n",
+    "                        validation_data=val_data_creator,\n",
+    "                        validation_steps=150 // batch_size,\n",
+    "                        data_config={\"mean\": mean, \"std\": std, \"batch_size\": batch_size, \"k\": k, \"i\": i, \"K-fold\": True},\n",
```

Comment:
Why does fit has a data_config option? Seems it would be expanded to the config of the estimator. What's the recommended usage? Would there be cases that fit with different configs? @yangw1234

Reply:
Also the batch size issue can be updated after this PR is finished https://github.com/intel-analytics/analytics-zoo/pull/3432